File: Java/AlgoExpert/Dynamic Programming/Medium/Number of Ways to Traverse Graph.java
import java.util.*;

class Program {

	public int numberOfWaysToTraverseGraph(int width, int height) {
		// Write your code here.
		int[][] grid = new int[height][width];

		for (int i = 0; i < height; i++) {
			for (int j = 0; j < width; j++) {
				if (i == 0 || j == 0) {
					grid[i][j] = 1;
					continue;
				}
				int above = grid[i - 1][j];
				int left = grid[i][j - 1];
				grid[i][j] = above + left;
			}
		}

		return grid[height - 1][width - 1];
	}
}

File: Python/AlgoExpert/Dynamic Programming/Medium/numberOfWaysToTraverseGraph.py
def numberOfWaysToTraverseGraph(width, height):
    # Write your code here.
    grid = [[0 for i in range(width)] for j in range(height)]
    grid[0][0] = 1

    for i in range(height):
        for j in range(width):
            if i - 1 >= 0:
                grid[i][j] += grid[i - 1][j]
            if j - 1 >= 0:
                grid[i][j] += grid[i][j - 1]

    return grid[-1][-1]
    
